src/Frontend/Middleware.hs

```diff
@@ -37,11 +37,14 @@ catch404 devMode app req cont = app req $ \resp -> cont $ f resp
         else responseBuilder status headers builder
       where
         status  = responseStatus resp
-        headers = responseHeaders resp
         builder = Builder.byteString . cs
                 . (`runReader` whereToGetTheLangValue) . renderTextT . toHtml
                 $ PublicFrame Page404 [] devMode
 
+        setContentType ("Content-Type", "text/plain") = ("Content-Type", "text/html;charset=utf-8")
```

Comment:
misleading name (even if local).  you are setting the content type *encoding*, not the *type*.  i'll change it to something.